File: Sep01/Troy/Q2.java
class Solution {
    public int search(int[] nums, int target) {
        int len = nums.length;
        if (len == 0) return -1;
        int start = 0;
        int end = len - 1;
        while(start <= end) {
            int middle = (end - start) / 2 + start;
            if (nums[middle] == target) return middle;
            if (nums[middle] > target) end = middle - 1;
            if (nums[middle] < target) start = middle + 1;
        }
        return -1;
    }
}

File: 2018/Sep01/Carl/Q2.py
from math import floor


class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        length = len(nums)
        if length == 0:
            return -1
        start = 0
        end = length-1
        while(start <= end):
            mid = floor((start+end)/2)
            if nums[mid] == target:
                return mid
            elif target > nums[mid]:
                start = mid + 1
            else:
                end = mid - 1
        return -1


S = Solution()
print(S.search([1, 2, 3, 4, 5], 5))
